### DISCUSSION TITLE
question

### DISCUSSION THREAD

**Discussion by leuamS** [86Box/86Box#2019] [Ideas]
can we have muticore compatibly?

**Comment by dhrdlicka** [comment]
We already use separate threads for certain components like network, audio or video. However, this is not really feasible for CPU emulation (which is what you likely mean), as it makes no sense to execute a single stream of instructions in parallel across multiple threads.

**Reply by leuamS** [reply]
OK, I just wanted a playable Slot 2 experience

**Reply by leuamS** [reply]
But dolphin can run on 2 cores and the cpus it emulates are 1 core

**Reply by ts-korhonen** [reply]
Just like @dhrdlicka explained, we already use multiple threads (that can run in separate cores - it's up to OS to decide where threads are run) in similar manner that is done in Dolphin.

It is so similar that the answer to this very question is the same as in [Dolphin FAQ document:](https://dolphin-emu.org/docs/faq/#couldnt-dolphin-use-more-my-cpu-cores-go-faster).
Here is a quote from answer to question; "Couldn't Dolphin use more of my CPU cores to go faster?"
> In Dolphin, the only demanding tasks that can run well in parallel are the CPU, the GPU and the DSP. Breaking up any of these tasks into smaller tasks just to run it on more cores is very likely to make the program slower. That's why Dolphin only runs on 3 cores and won't use all of your 4 or 6 cores CPU.

The same applies to 86Box, but when emulating the x86 processors of the era, the CPU is the biggest workload - on Dolphin I imagine that GPU and DSP workloads are proportionally closer to CPU workload and thus more visible.

**Reply by leuamS** [reply]
ok
